scripts/gulp/frontend-shared.js

```diff
@@ -15,7 +15,7 @@ const buildFrontendSharedJs = () => {
 
   return (
     gulp
-      .src('frontend-shared/src/*')
+      .src('frontend-shared/src/**/!(*-test).js')
```

Comment:
I was a little confused about the relationship of this to this set of changes until I read your commit message. Might I suggest breaking this out into a separate commit? This change doesn't feel like it belongs with the PascalCasing/component structure changes.

Reply:
I made a change with the same functional effect in https://github.com/hypothesis/client/pull/2951 following Kyle's feedback. However I used a negative glob in a glob array in order to exclude _all_ files in `test` directories and also because the "negative glob in a glob array" approach is documented at https://gulpjs.com/docs/en/getting-started/explaining-globs/, whereas this more advanced minmatch syntax is not.